codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/protocols/serialize/ValueExpression.kt

```diff
@@ -22,4 +22,11 @@ sealed class ValueExpression {
         is Reference -> name
         is Value -> "&$name"
     }
+
+    override fun toString(): String {
+        return when (this) {
+            is Reference -> name
+            is Value -> name
+        }
+    }
```

Comment:
Do you need the `when` at all? I would've thought no because of line 11.
Why don't we need a `&` for the `Reference` case? 🤔

Reply:
Good catch! 
The `Reference` case can either be a `&variable_name` or simply `variable_name` with the implicit understanding that the variable points to a reference, so we should avoid adding an additional `&`.

Reply:
Basically, I want references to be rendered as references and values to be rendered as value if no other method has been explicitly called (i.e. `asValue`/`asRef`).